src/compat/posix/proc/atexit/atexit.c

```diff
@@ -0,0 +1,38 @@
+/**
+ * @file
+ * @brief
+ *
+ * @date 05.8.2020
+ * @author  Erick Cafferata
+ */
+
+#include <kernel/task/resource/atexit.h>
+#include <limits.h>
+
+struct task_atexit *list;
+
+static int slot;
```

Comment:
You have not to use global variables.
Update 'struct task_atexit' adding int slot, and use if for all date